mlflow/types/schema.py

```diff
@@ -122,34 +122,267 @@ def from_numpy_type(cls, np_type):
         return next((v for v in cls._member_map_.values() if v.to_numpy() == np_type), None)
 
 
+ArrayType = TypeVar("ArrayType", bound="Array")
+ObjectType = TypeVar("ObjectType", bound="Object")
+
+
+@experimental
+class Property:
+    """
+    Specification used to represent a json-convertible object property.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        dtype: Union[DataType, ArrayType, ObjectType, str],  # pylint: disable=redefined-builtin
```

Comment:
Does `DataType` contains `str`?

Reply:
Here `str` means the name of a DataType, we convert it into DataType[dtype] if applicable. The property dtype should only be Union[DataType, ArrayType, ObjectType].